src/main/java/at/hannibal2/skyhanni/features/inventory/ChestValue.kt

```diff
@@ -147,31 +147,39 @@ object ChestValue {
 
     // TODO: Avoid Ordinal
     private fun MutableList<Renderable>.addButton() {
-        addButton(
-            prefix = "§7Sorted By: ",
-            getName = SortType.entries[config.sortingType.ordinal].longName,
+
+        addRenderableButton<SortingTypeEntry>(
+            label = "Price Sorting",
+            current = config.sortingType,
+            getName = { it.displayName },
             onChange = {
-                config.sortingType = SortingTypeEntry.entries[(config.sortingType.ordinal + 1) % 2]
+                config.sortingType = it
                 update()
             },
+            universe = SortingTypeEntry.entries,
         )
 
-        addButton(
-            prefix = "§7Value format: ",
-            getName = FormatType.entries[config.formatType.ordinal].type,
+        addRenderableButton<NumberFormatEntry>(
+            label = "Value Format",
+            current = config.formatType,
+            getName = { it.displayName },
             onChange = {
-                config.formatType = NumberFormatEntry.entries[(config.formatType.ordinal + 1) % 2]
+                config.formatType = it
                 update()
             },
+            universe = NumberFormatEntry.entries,
         )
 
-        addButton(
-            prefix = "§7Display Type: ",
-            getName = DisplayType.entries[if (config.alignedDisplay) 1 else 0].type,
+        // TODO boolean support
```

Comment:
Boolean support for what exactly? (Again could use some more descriptive stuff here imo)

Reply:
stuff i wanna do in this pr, again

Reply:
i dont think ill do this in this pr, ive added a better todo description though